### PR TITLE
RT-12219: LevelCreditApiClient first part

### PR DESCRIPTION
**Pull request by yurez**

### CODE REVIEW

File: src/LoggableBehavior/DefaultLogHandler.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace LevelCredit\LevelCreditApi\LoggableBehavior;
```
Comment (tatyana-shmatchenko) [inline src/LoggableBehavior/DefaultLogHandler.php:3]: Let's discuss this namespace.

**Review by tatyana-shmatchenko — CHANGES_REQUESTED** [review]
Looks great @yurez !
Really like it :+1: 
Let's discuss logs and doctypes.

File: src/LevelCreditApiClient.php [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    protected $serializer;
+
+    /**
+     * @param Serializer $serializer
+     * @param string $clientId
+     * @param string $clientSecret
+     * @param string $baseUri
+     */
+    public function __construct(
+        Serializer $serializer,
```
Comment (caryp) [inline src/LevelCreditApiClient.php:89]: Why supply a serializer in the constructor if you are just going to default it?  Why not use an accessor method to set this if you want to use something non-default?

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "levelcredit/levelcredit-api-php",
+  "description": "LevelCredit Distributed API Client Library for PHP",
```
Comment (caryp) [inline composer.json:3]: please remove "Distributed"

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  "minimum-stability": "stable",
+  "license": "MIT",
```
Comment (caryp) [inline composer.json:5]: I am checking on what this license should be with Shivan.  Good enough for now, but might need to change.

**Review by caryp — CHANGES_REQUESTED** [review]
Looking very good, @yurez ! I only did a high-level review. A couple comments added.  Also please add at least enough documentation in the readme to tell me how to install and make a simple request.  Ideally a way to run the tests too.

File: src/Enum/UserStatus.php [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace LevelCredit\LevelCreditApi\Enum;
+
+class UserStatus
+{
+    public const INVITE = 'invite';
```
Comment (tatyana-shmatchenko) [inline src/Enum/UserStatus.php:7]: Should we add any other statuses?

File: src/Logging/DefaultLogHandler.php [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        . "\n{res_body} \n{error}";
+
+    private const WOUT_ANY = "[{ts}]{hostname} - \"{method} {target} \" {code} {phrase} "
+        . "{res_header_Content-Length} \n{error}";
+
+    private const PRIVATE_HEADERS_REPLACEMENT = [
+        '/authorization/i' => 'X-REMOVED-AUTHORIZATION-X',
+    ];
+
+    private const PRIVATE_FIELDS_REPLACEMENT = [
```
Comment (tatyana-shmatchenko) [inline src/Logging/DefaultLogHandler.php:25]: Please add client_secret.

File: src/Serializer/Serializer.php [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+use LevelCredit\LevelCreditApi\Model\Response\BaseResponse;
+use LevelCredit\LevelCreditApi\Model\Response\CollectionResponse;
+use LevelCredit\LevelCreditApi\Model\Response\EmptyResponse;
+use LevelCredit\LevelCreditApi\Model\Response\ErrorCollection;
+use LevelCredit\LevelCreditApi\Model\Response\ResourceResponse;
+use Psr\Http\Message\ResponseInterface;
+
+class Serializer
+{
+    protected const FAILED_STATUS_ENTRYPOINT = 400;
```
Comment (tatyana-shmatchenko) [inline src/Serializer/Serializer.php:19]: ENTRYPOINT -> ENTRY_POINT

**Review by tatyana-shmatchenko — APPROVED** [review]
Great job :+1:

File: src/Enum/UserStatus.php (hunk 5 above)
Comment (yurez) [reply src/Enum/UserStatus.php:7]: We are not supported right now other statuses in the api.

**Comment by yurez** [comment]
@caryp it's ready for your review, readme file will be added in next sprint

**Comment by caryp** [comment]
@yurez I reviewed 2days ago. please address my comments. thank you

**Comment by tatyana-shmatchenko** [comment]
Discussed with @caryp on the standup that this is approved! Merging.

**Pull request merged**